tokio/src/sync/mpsc/list.rs

```diff
@@ -252,6 +252,20 @@ impl<T> Rx<T> {
         }
     }
 
+    /// Peeks at the next value on the queue, if any.
+    pub(crate) fn peek(&mut self) -> Option<block::Read<T>> {
```

Comment:
What does this function do? It's name is `peek`, which makes it sounds like it does _not_ consume the next value, but the implementation looks like it consumes the next value.

Reply:
The fn `peek` is modeled after `pop` above it.  The input is mutable due to the signature `try_advancing_head(&mut self)`.  The list maintains an index into the current message `self.index` whose contents are stored in finite-sized blocks organized as a linked list, with head `self.head`.  If the head block cannot contain the message for `self.index` then `self.head` should be advanced.  Finally, because `peek` doesn't mutate `self.index` -- the next message remains the same before and after the call.  If/when `pop` is subsequently called; the `self.head` pointer has already been updated and at that time `reclaim_blocks` can be called because the caller now has ownership of the data.

Reply:
peel and try_advancing_head have been removed.